javaparser-core/src/main/java/com/github/javaparser/ast/type/ArrayType.java

```diff
@@ -118,6 +118,12 @@ public static Type wrapInArrayTypes(Type type, List<ArrayBracketPair>... arrayBr
                 }
             }
         }
+        // The above code concatenates the arrayBracketPairLists, because they represent different
+        // parts of the array rather than two sources of information about the same parts of the
+        // array.
+        if (arrayBracketPairLists.length > 0) {
+            type.setArrayAnnotations(arrayBracketPairLists[0]);
```

Comment:
Why does this ignore everything but the outermost set of array brackets?
Does this pass tests that you have written that contain multi-dimensional arrays with annotations on various levels?